CMakeLists.txt

```diff
@@ -677,6 +694,54 @@ elseif(APPLE)
         ${CMAKE_COMMAND} -E remove_directory
         ${SDK_INSTALL_DIR}
     )
+
+# Need to make Mac-specific changes to the dylib install_name for
```

Comment:
That's a b2 job, not an SDK build one.

According to http://boost.2283326.n4.nabble.com/Need-to-add-quot-install-name-quot-to-dylibs-on-OS-X-td2692930.html

http://www.boost.org/build/doc/html/bbv2/overview/builtins/features.html#bbv2.reference.features.dll-path should do the same job in a less messy way.

Reply:
There are other people who have noted that b2 cannot handle this job.
Or at least, it was broken in 1.45 and not working until 1.62 (and we are using 1.59).
[https://stackoverflow.com/questions/3431619/how-to-force-boost-to-use-rpath](url=https://stackoverflow.com/questions/3431619/how-to-force-boost-to-use-rpath)

Reply:
To be clear, I wanted to use b2 to take care of the install_name paths.
I spent several days in June trying to get it to work.
I was unable to get b2 to do what was needed.

I had found the same online links that adrianbroher listed.
I tried it before he suggested it.
I tried it after he suggested it.
I just retried it now.
Still doesn't work (on the Mac).

There is a solution that works right now, which is what I committed:
having cmake use install_name_tool to modify the shared libraries after b2 is finished.
Unless someone can point out a working solution using v1.59 of boost, I don't think it is possible to make this a "b2 job".

Reply:
This is a regression in boost.
Can we use the working solution and mark it as compensating for the regression in boost that should be reverted when we adopt boost v1.62?

Reply:
Well, either that or just do the switch to boost 1.62 now. One of these *have* to be done before the 0.4.8 release to be able to solve the issues with the logger on OSX.

I'm fine with either one (or any other solution anyone can come up with), as long as this gets resolved in the foreseeable future.

Reply:
1.62, or 1.63?
Only reason I ask is this PR ( https://github.com/freeorion/freeorion/pull/1775 ) that talks about Fedora users having problems with 1.62 or lower.

Reply:
Whatever works. If 1.62 has issues on other platforms, then switch to 1.63. As long as we get this mess cleaned up finally... 😉

Reply:
I'll check with 1.62 and 1.63, to see if either (or both) work on the Mac like they are supposed to.
After getting that info, wiser heads can decide on the best course of action.

Reply:
Okay. I really think this (using install_name_tool as part of the build script) is the only viable solution. While it would be nice to use b2, it isn't designed to do that (for the Mac). Changing the boost version does not fix that.

(While it is possible to do it in theory, it would require building each Boost library one at a time. Throwing away the current build scripts, and trying to guess the correct settings of each library, is not what I would consider a proper solution.)

Reply:
So, that means this issue hasn't been resolved even with boost 1.63? That's unfortunate.

@adrianbroher, as your objection to resort to the hack using `install_name_tool` is preventing us from using that fix, and the solutions you suggested apparently don't work, how do you suggest to proceed?

I don't want to do anything that you perceive as subverting all the effort you've put into the SDK, but we *need* to fix this issue. Honestly, I'm out of ideas... 😕

Reply:
@Vezzra, I get the feeling that there is nothing to "fix", that Boost does not support adjusting the install_path (on OSX).

They felt the need to add this sentence to the documentation: <i>"Note that on Mac OSX, the paths are unconditionally hardcoded by the linker, and it is not possible to disable that behaviour."</i> I suspect this means that the only options are to pass-through a flag (with the path) to the linker, or to modify the install_paths after the libraries are built. The former only works if we are using b2 to make a single library, and the latter is what I submitted (using install_name_tool in the cmake script).

Reply:
Ok, I see. @adrianbroher, in light of these findings, how do you suggest to proceed?

> The former only works if we are using b2 to make a single library

I assume that means building only one of the boost libraries by a single invocation of the b2 build system? If yes, would a possible fix be to change the cmake scripts so that each boost library gets built by separate invocations of the b2 build system?

That said, that solution isn't really better than using install_name_tool, IMO. However, if @adrianbroher prefers it, I can live with that. @adrianbroher?

Reply:
I was thinking of the linker flag "-install_path", where the absolute path can be defined. Since that will include the name of the library, it would seem to only work 1 library at a time.

I think having b2 make one library at a time is a rather ugly solution. Some of the boost libraries depend on others, so we would have to reverse engineer a lot of stuff that Boost is taking care of automatically.

The original request was, "That's a b2 job, not an SDK build one." Everything I can find suggests that b2 [u]cannot[/u] do that job (with OSX). Someone would need to point out detailed instructions on how b2 can do it, in order for me to make the requested change.

Reply:
> I was thinking of the linker flag "-install_path", where the absolute path can be defined. Since that will include the name of the library, it would seem to only work 1 library at a time.

Ah, ok, I understand.

> Some of the boost libraries depend on others, so we would have to reverse engineer a lot of stuff that Boost is taking care of automatically.

Ugh. That indeed sounds messy. So, not going to happen.

That means the only working solution still on the table is using `install_name_tool`. Unfortunately, lacking any further feedback from @adrianbroher, we're kind of stuck. Of course, we could just overrule his objections, but I really, *really* don't want to do this. :frowning_face: